src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -207,12 +207,33 @@ def build_summary_data(
                 group__in=([group for group in regressed_or_escalated_groups]),
                 type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_ESCALATING.value),
             )
-            if regressed_or_escalated_groups_today:
-                for activity in regressed_or_escalated_groups_today[:4]:
-                    if activity.type == ActivityType.SET_REGRESSION.value:
-                        project_ctx.regressed_today.append(activity.group)
+
+            deduped_groups_by_activity_type: dict[str, set] = {
+                "regressed": set(),
+                "escalated": set(),
+            }
+            for activity in regressed_or_escalated_groups_today:
+                if activity.type == ActivityType.SET_ESCALATING.value:
+                    # if a group is already in the regressed set but we now see it in escalating, remove from regressed and add to escalating
+                    # this means the group regressed and then later escalated, and we only want to list it once
+                    if activity.group in deduped_groups_by_activity_type["regressed"]:
+                        deduped_groups_by_activity_type["regressed"].remove(activity.group)
+                    deduped_groups_by_activity_type["escalated"].add(activity.group)
+                else:
+                    if (
+                        activity.type == ActivityType.SET_REGRESSION.value
+                        and activity.group not in deduped_groups_by_activity_type["escalated"]
+                    ):
+                        deduped_groups_by_activity_type["regressed"].add(activity.group)
+
+            if deduped_groups_by_activity_type:
+                for activity_type, groups in deduped_groups_by_activity_type.items():
+                    if activity_type == "regressed":
+                        for group in list(groups)[:4]:
```

Comment:
why are we skipping the last 4 items in the list?

Reply:
This gets the first 4 items, we have to limit the results for the notification otherwise the payload could be too large and it'd fail to send.